src/index/scanners/maxsim.rs

```diff
@@ -90,191 +90,213 @@ impl SearchBuilder for MaxsimBuilder {
-                                accu_set.extend(reranker.by_ref().take(maxsim_refine as _));
-                                let (rough_iter, accu_iter) = reranker.finish();
-                                accu_set.extend(accu_iter.map(accu_map));
-                                rough_set.extend(rough_iter.into_iter().map(rough_map));
+                let vectors = original_vectors
+                    .clone()
+                    .into_iter()
+                    .map(|vector| RandomProject::project(vector.as_borrowed()));
+                Box::new(vectors.into_iter().zip(original_vectors).map(
+                    |(vector, original_vector)| {
+                        let (results, estimation_by_threshold) = match options.io_search {
+                            Io::Plain => maxsim_search::<_, Op>(
+                                index,
+                                vector.clone(),
+                                options.probes.clone(),
+                                options.epsilon,
+                                maxsim_threshold,
+                                bump,
+                                make_h1_plain_prefetcher.clone(),
+                                make_h0_plain_prefetcher.clone(),
+                            ),
+                            Io::Simple => maxsim_search::<_, Op>(
+                                index,
+                                vector.clone(),
+                                options.probes.clone(),
+                                options.epsilon,
+                                maxsim_threshold,
+                                bump,
+                                make_h1_plain_prefetcher.clone(),
+                                make_h0_simple_prefetcher.clone(),
+                            ),
+                            Io::Stream => maxsim_search::<_, Op>(
+                                index,
+                                vector.clone(),
+                                options.probes.clone(),
+                                options.epsilon,
+                                maxsim_threshold,
+                                bump,
+                                make_h1_plain_prefetcher.clone(),
+                                make_h0_stream_prefetcher.clone(),
+                            ),
+                        };
+                        let (mut accu_set, mut rough_set) = (Vec::new(), Vec::new());
+                        if maxsim_refine != 0 && !results.is_empty() {
+                            match options.io_rerank {
+                                Io::Plain => {
+                                    let prefetcher = PlainPrefetcher::<_, BinaryHeap<_>>::new(
+                                        index,
+                                        results.into(),
+                                    );
+                                    let mut reranker = rerank_index::<Op, _, _>(
```

Comment:
Why don't apply filter for maxsim search?

Reply:
Added.